plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php

```diff
@@ -2065,9 +2065,14 @@ function ( $collection_args, $query ) {
 					if ( isset( $location['type'] ) && 'product' === $location['type'] ) {
 						$product_references = array( $location['sourceData']['productId'] );
 					}
+
 					if ( isset( $location['type'] ) && 'cart' === $location['type'] ) {
 						$product_references = $location['sourceData']['productIds'];
 					}
+
+					if ( isset( $location['type'] ) && 'order' === $location['type'] ) {
+						$product_references = $this->get_product_ids_from_order( $location['sourceData']['orderId'] ?? null );
```

Comment:
We should fallback to `0` here since the `get_product_ids_from_order` expects an `int`

Reply:
I don't have a strong opinion here, so I added fallback to 0.